doc/Language/traits.pod6

```diff
@@ -108,6 +108,27 @@ would be a way of adding functionality which cuts across class and role
 hierarchies, or can even be used to add behaviors to independently defined
 routines.
 
+=head2 X<C<is implementation-detail>,traits> trait
+
+Note: this trait is available in Rakudo compiler starting from 2020.05 release.
+
+The C<is> with C<implementation-detail> argument is used by Raku
+language implementations and module authors to mark particular
+routines (including methods) as not meant to be a public API.
+
+Applying this trait makes the C<is-implementation-detail> method on
+L<Code|/type/Code> to return C<True>.
+
+    my &fail-routine = &fail;
+    unless &fail-routine.is-implementation-detail {
+        say "It is safe to use &fail routine in your modules!";
+    }
+
+    sub PRIVATE-CALCULATION is implementation-detail { #`(Not safe to rely on this) }
+    if &PRIVATE-CALCULATION.is-implementation-detail {
+        say "You better not to rely on &PRIVATE-CALCULATION unless you really know what you are doing";
+    }
+
 =end pod
 
```

Comment:
Maybe also add something about what it implies; for instance, not showing up in method lookups (if that's the case)

Reply:
Added.